src/analytic_engine/src/compaction/runner/remote_runner.rs

```diff
@@ -0,0 +1,48 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use async_trait::async_trait;
+use cluster::ClusterRef;
+use generic_error::BoxError;
+use snafu::ResultExt;
+
+use crate::{
+    compaction::runner::{CompactionRunner, CompactionRunnerResult, CompactionRunnerTask},
+    instance::flush_compaction::{ConvertCompactionTaskResponse, RemoteCompact, Result},
+};
+
+pub struct RemoteCompactionRunner {
+    pub cluster: ClusterRef,
+}
+
+#[async_trait]
+impl CompactionRunner for RemoteCompactionRunner {
```

Comment:
It may be good to place all remote compaction logic into this as described above.